src/dlr_tvm.cc

```diff
@@ -198,3 +207,90 @@ void TVMModel::UseCPUAffinity(bool use) {
     LOG(INFO) << "CPU Affinity is disabled";
   }
 }
+
+bool TVMModel::HasMetadata() const {
+  if (this->metadata != nullptr) {
+    return true;
+  }
+  return false;
+}
+
+void TVMModel::GetOutputNames(char**& names) const {
+  names = (char**) malloc(this->num_outputs_);
+  std::string name;
+
+  for(int i = 0; i < this->num_outputs_; i++) {
+    name = this->metadata["Model"]["Outputs"][i]["name"];
+    names[i] = (char *)malloc((name.size() + 1)*sizeof(char));
+    strcpy(names[i], name.c_str());
+  }
+}
+
+void TVMModel::GetInputNames(char**& names) const {
+  names = (char**) malloc(this->num_inputs_);
+  std::string name;
+
+  for(int i = 0; i < this->num_inputs_; i++) {
+    name = this->metadata["Model"]["Inputs"][i]["name"];
+    names[i] = (char *)malloc((name.size() + 1)*sizeof(char));
+    strcpy(names[i], name.c_str());
+  }
+}
+
+void TVMModel::GetOutputIndex(const char* name, int* index) const {
+  char** output_names;
+  this->GetOutputNames(output_names);
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    if (strcmp(name, output_names[i]) == 0) {
+      idx = i;
+      break;
+    }
+  } 
+  *index = idx;
+}
+
+void TVMModel::GetInputIndex(const char* name, int* index) const {
+  char** input_names;
+  this->GetInputNames(input_names);
+  int idx = -1;
+
+  for (int i = 0; i < this->num_outputs_; i++) {
+    if (strcmp(name, input_names[i]) == 0) {
+      idx = i;
+      break;
+    }
+  } 
+  *index = idx;
+}
+
+void TVMModel::GetOuputByName(const char* name, float* out) {
+  int* output_index;
+  this->GetOutputIndex(name, output_index);
+  if (*output_index >= 0) {
+    this->GetOutput(*output_index, out);
+  }
+}
+
+void TVMModel::GetInputDataTypes(char**& dtypes) const {
```

Comment:
You can use `GetInputType(index)` to get input type from compiled model json. `GetInputType(index)` was added to neo-ai/tvm graph_runtime https://github.com/neo-ai/tvm/commits/r20200204 (update 3rdparty/tvm commit in neo-ai-dlr)

Reply:
In that case we should merge https://github.com/neo-ai/neo-ai-dlr/pull/174 first since latest neo-ai/tvm requires other changes in that PR for CI and Cmake

Reply:
We could wait on input types related changes and release them in a separate PR. I would rather release outpu related changes  with DLR 1.2 @trevor-m @apivovarov